### ISSUE TITLE
Audit and reorganize tests (in astropy.coordinates)

### ISSUE THREAD

**Issue by adrn** [astropy/astropy#6577]
I added `astropy.coordinates` to the title because I want to look at that in particular, but I do think we need a general review of tests in the full package. I'm planning to start looking at this in a breakout during the coordination meeting this week.

**Comment by mhvk** [comment]
Yes, good idea! Probably also good to first summarize how you would like to re-organize...

**Comment by adrn** [comment]
The main thing I had in mind is that there are a bunch of tests in `test_frames.py` that could and should be enumerated over all frames. But I'll certainly lay out a plan here before doing anything drastic!
